cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1207,27 +1207,43 @@ impl FromStr for CatalogPackage {
     ///
     ///     attribute_path ::= <install_id> | <attribute_path_rest>.<install_id>
     ///     attribute_path_rest ::= <identifier> | <attribute_path_rest>.<identifier>
-    ///     install_id ::= <identifier>
+    ///     install_id ::= <identifier> | @<identifier>
     ///
     ///     version ::= <string> # interpreted as semver or plain version by the resolver
     /// ```
-    /// Todo: this does currently _not_ handle any more pathological cases like
-    ///  - `@` in the version string (the last `@` is the delimiter)
     fn from_str(descriptor: &str) -> Result<Self, ManifestError> {
-        let (attr_path, version) = match descriptor.split_once('@') {
-            Some((attr_path, version)) if !version.is_empty() => {
-                (attr_path.to_string(), Some(version.to_string()))
-            },
-            Some(_) => {
+        fn split_version(haystack: &str) -> (usize, Option<&str>) {
+            let Some(version_at) = haystack.find('@') else {
+                return (haystack.len(), None);
+            };
+
+            if version_at == 0 {
+                return (haystack.len(), None);
+            }
+
+            if &haystack[version_at - 1..version_at] == "." {
+                let (next_version_at, next_version) = split_version(&haystack[version_at + 1..]);
```

Comment:
I don't think we care too much about the recursive case, but if we do we should test and comment this (e.g. it's confusing that next_version_at is for the next `@` but next_version is actually the final version)

Reply:
replaced the recursion with a loop, i hope this is a bit clearer now.